### DISCUSSION TITLE
How to change element content in sciter.js?

### DISCUSSION THREAD

**Discussion by kitepad** [c-smile/sciter-js-sdk#97] [Ideas]
Hi, I can change button's text dynamically using zepto.js (refer by samples/$).
Sciter.JS version = latest.

below source code is right : `<html>
![image](https://user-images.githubusercontent.com/28701482/116950143-bf649f00-acb6-11eb-8dd9-0c85f0173122.png)

I want use sciter.js only (not depend on zepto.js), but I failed many times.
Would you give me equivalent solution without any dependencies.

Thanks.

**Comment by c-smile** [comment]
```html
<html>
    <head>
        <title>Test</title>
        <style></style>
        <script>

var counter = 0;

document.$("button").on("click", function() {
  this.innerHTML = `clicked ${++counter} times`;
});

        </script>
    </head>
    <body>

<button>Click me</button>

    </body>
</html>
```
This above:
```  this.innerHTML = `clicked ${++counter} times`;```
should be 
```  this.innerText = `clicked ${++counter} times`;```
but it is a bug in innerText implementation for buttons. My pardon, fixed here will appear in the next build.

**Comment by kitepad** [comment]
Thanks!   I'm researching htmx (htmx.org) and want to modify it and use with sciter.js. I think it's a better approach to reduce event handler efforts and avoid many errors especially for local desktop UI with remote backend application architecture. 

But htmx is relied on xmlhttprequest, it's failed for my first try.

**Comment by c-smile** [comment]
> But htmx is relied on xmlhttprequest,

xmlhttprequest is quite outdated, Sciter supports [Fetch API](https://medium.com/beginners-guide-to-mobile-web-development/the-fetch-api-2c962591f5c) instead.

But I think in case of Sciter and desktop application with local backend it is more convenient to just add one native gate-function 

```backend.exec("get"|"put"|"delete", "path" [, params ])```

**Comment by kitepad** [comment]
I like your idea. My expectation is use sciter.js to develop android application, and have a gateway or façade to talk with remote backend. This is similar as BFF pattern, My sciter.js app is separate into 3 layer : UI (user interface ) layout layer and UI (user interaction) logic layer in customer device (laptop or android mobile, package by sciter.js quark), business logic or  data service layer are running on remote. with exposed APIs.
